scripts/idl/zapxml/handlers/handlers.py

```diff
@@ -0,0 +1,617 @@
+            return BaseHandler(self.context, handled=HandledDepth.SINGLE_TAG)
+        elif name.lower() == 'description':
+            return BaseHandler(self.context, handled=HandledDepth.ENTIRE_TREE)
+        else:
+            return BaseHandler(self.context)
+
+    def EndProcessing(self):
+        self._cluster.events.append(self._event)
+
+
+class AttributeHandler(BaseHandler):
+    """Handles /configurator/cluster/attribute elements."""
+
+    def __init__(self, context: Context, cluster: Cluster, attrs):
+        super().__init__(context)
+        self._cluster = cluster
+        self._attribute = AttrsToAttribute(attrs)
+
+    def GetNextProcessor(self, name: str, attrs):
+        if name.lower() == 'access':
+            if 'modifier' in attrs:
+                if attrs['modifier'] != 'fabric-scoped':
+                    raise Exception("UNKNOWN MODIFIER: %s" % attrs['modifier'])
+                self._attribute.tags.add(AttributeTag.FABRIC_SCOPED)
+            else:
+                role = AttrsToAccessPrivilege(attrs)
+
+                if attrs['op'] == 'read':
+                    self._attribute.readacl = role
+                elif attrs['op'] == 'write':
+                    self._attribute.writeacl = role
+                else:
+                    logging.error("Unknown access: %r" % attrs['op'])
+            return BaseHandler(self.context, handled=HandledDepth.SINGLE_TAG)
+        elif name.lower() == 'description':
+            return AttributeDescriptionHandler(self.context, self._attribute)
+        else:
+            return BaseHandler(self.context)
+
+    def HandleContent(self, content: str):
+        # Content generally is the name EXCEPT if access controls
+        # exist, in which case `description` contains the name
+        content = content.strip()
+        if content and not self._attribute.definition.name:
+            self._attribute.definition.name = content
+
+    def EndProcessing(self):
+        if self._attribute.definition.name is None:
+            raise Exception("Name for attribute was not parsed.")
+
+        self._cluster.attributes.append(self._attribute)
+
+
+class StructHandler(BaseHandler, IdlPostProcessor):
+    """ Handling /configurator/struct elements."""
+
+    def __init__(self, context: Context, attrs):
+        super().__init__(context)
+
+        # if set, struct belongs to a specific cluster
+        self._cluster_codes = set()
+        self._struct = Struct(name=attrs['name'], fields=[])
+        self._field_index = 0
+        # The following are not set:
+        #    - tag not set because not a request/response
+        #    - code not set because not a response
+
+        self._is_fabric_scoped = (attrs.get('isFabricScoped', "false").lower() == 'true')
+
+    def GetNextProcessor(self, name: str, attrs):
+        if name.lower() == 'item':
+            data_type = DataType(
+                name=attrs['type']
+            )
+
+            # TODO: handle isFabricSensitive
```

Comment:
Please file an issue and reference here with `TODO(#issue):`

Reply:
Filed an issue: this is generally for the IDL format - it did not consider isFabricSensitive at all.